src/java.base/share/classes/java/net/Socket.java

```diff
@@ -379,25 +379,24 @@ public Socket(InetAddress address, int port, InetAddress localAddr,
      * loopback interface. </p>
      * <p>
      * If the stream argument is {@code true}, this creates a
-     * stream socket. If the stream argument is {@code false}, it
-     * creates a datagram socket.
+     * stream socket. Only stream socket creation is allowed. If the stream
+     * argument is {@code false}, then this constructor throws
+     * {@code IllegalArgumentException}.
      * <p>
      * If the application has specified a {@linkplain SocketImplFactory client
      * socket implementation factory}, that factory's
      * {@linkplain SocketImplFactory#createSocketImpl() createSocketImpl}
      * method is called to create the actual socket implementation. Otherwise
      * a system-default socket implementation is created.
-     * <p>
-     * If a UDP socket is used, TCP/IP related socket options will not apply.
      *
      * @param      host     the host name, or {@code null} for the loopback address.
      * @param      port     the port number.
      * @param      stream   a {@code boolean} indicating whether this is
```

Comment:
I think the parameter description has to be changed as it no longer indicates if the socket is a stream or datatgram socket. I think you can replace with something simple, like must be true, false is not allowed.

Reply:
Done.